Remotion/Data/DomainObjects/Persistence/Rdbms/Parameters/ObjectIDDataParameterDefinitionFactory.cs

```diff
@@ -0,0 +1,78 @@
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using Remotion.Data.DomainObjects.Persistence.Configuration;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Model.Building;
+using Remotion.Data.DomainObjects.Queries;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.Persistence.Rdbms.Parameters;
+
+/// <summary>
+/// Can create <see cref="IDataParameterDefinition"/> instances that can handle <see cref="ObjectID"/> parameter values.
+/// </summary>
+public class ObjectIDDataParameterDefinitionFactory : IDataParameterDefinitionFactory
+{
+  public IStorageSettings StorageSettings { get; }
+  public StorageProviderDefinition StorageProviderDefinition { get; }
+  public IStorageTypeInformationProvider StorageTypeInformationProvider { get; }
+  public IDataParameterDefinitionFactory NextDataParameterDefinitionFactory { get; }
+
+  public ObjectIDDataParameterDefinitionFactory (StorageProviderDefinition storageProviderDefinition,
+      IStorageTypeInformationProvider storageTypeInformationProvider,
+      IStorageSettings storageSettings,
+      IDataParameterDefinitionFactory nextDataParameterDefinitionFactory)
+  {
+    ArgumentUtility.CheckNotNull(nameof(storageProviderDefinition), storageProviderDefinition);
+    ArgumentUtility.CheckNotNull(nameof(storageTypeInformationProvider), storageTypeInformationProvider);
+    ArgumentUtility.CheckNotNull(nameof(storageSettings), storageSettings);
+    NextDataParameterDefinitionFactory = nextDataParameterDefinitionFactory;
+
+    StorageSettings = storageSettings;
+    StorageProviderDefinition = storageProviderDefinition;
+    StorageTypeInformationProvider = storageTypeInformationProvider;
+  }
+
+  public IDataParameterDefinition CreateDataParameterDefinition (QueryParameter queryParameter)
+  {
+    ArgumentUtility.CheckNotNull(nameof(queryParameter), queryParameter);
+
+    var objectID = queryParameter.Value as ObjectID;
+    return objectID != null
+        ? CreateDataParameterDefinition(objectID)
+        : NextDataParameterDefinitionFactory.CreateDataParameterDefinition(queryParameter);
+  }
+
+  private IDataParameterDefinition CreateDataParameterDefinition (ObjectID objectID)
+  {
+    var relatedStorageProviderDefinition = StorageSettings.GetStorageProviderDefinition(objectID.ClassDefinition);
+    return StorageProviderDefinition == relatedStorageProviderDefinition
+        ? CreateSameProviderRelationStoragePropertyDefinition()
+        : CreateCrossProviderRelationStoragePropertyDefinition();
+  }
```

Comment:
Stylistic bit:

I prefer using IF/ELSE instead of ternary operator for regular decisions on code. the main advantage is that most code coverage tools only work on a statement level so we can't check if both branches are covered. 

The other aspect is that ternary is meant for low-key decisions / simple values, not cases where control flow should be a prominent part of the system. I'm aawre that pattern matching, switch expressions and Linq dilute this concept further and further.

So, bascially: if you can put it in a single line and it's just a reult value decision (0 or 1 for instance), ternerary is okay. Using it for control flow, I try to stick to IF/ELSE because it adds some weight and space to the whole thing. 

But yeah, it's possible that missing code coverage is the main reason I don't like to use ternary for testable controlflow.